Makefile

```diff
@@ -16,7 +16,7 @@ IGVM_BUILD_SCRIPT := ./scripts/igvmmeasure/igvm.sh
 
 define compile_service
 	CGO_ENABLED=$(CGO_ENABLED) GOOS=$(GOOS) GOARCH=$(GOARCH) GOARM=$(GOARM) \
-	go build -ldflags "-s -w \
+	go build -ldflags "-linkmode external -extldflags '-static' -s -w \
```

Comment:
Let's do this in a separate PR, and ideally it should include only the agent, or if for all services we need to provide detailed docs on the dependencies or development environment required to build the services